src/main/java/de/retest/recheck/cli/subcommands/Diff.java

```diff
@@ -1,73 +1,169 @@
 
-	@Option( names = "--help", usageHelp = true, hidden = true )
-	private boolean displayHelp;
+	private static final PersistenceFactory persistenceFactory = new PersistenceFactory( getXmlTransformer() );
+	private static final GoldenMasterProviderImpl goldenMasterProvider =
+			new GoldenMasterProviderImpl( persistenceFactory.getPersistence() );
+
+	private static final String TEST_NAME = "Test - Golden Master Comparison";
+	private static final String SUITE_NAME = "Suite - Golden Master Comparison";
+
+	// TODO Temporarily it will be allowed to use the diff command for display of test reports
+	// TODO Will be removed in following version; instead, use the command "show"
+	@Parameters( arity = "1..2", description = GoldenMasterUtil.GOLDEN_MASTER_PARAMETER_DESCRIPTION )
+	private Path[] goldenMasterPath;
 
 	@Option( names = "--exclude", description = FilterUtil.EXCLUDE_OPTION_DESCRIPTION )
 	private List<String> exclude;
 
-	@Parameters( arity = "1", description = TestReportUtil.TEST_REPORT_PARAMETER_DESCRIPTION )
-	private File testReport;
+	@Option( names = "--help", usageHelp = true, hidden = true )
+	private boolean displayHelp;
 
-	@Override
-	public void run() {
-		if ( !PreCondition.isSatisfied() ) {
+	@Option( names = "--output", arity = "1",
+			description = "Save differences of Golden Masters to specified directory as test report." )
+	private Path directory;
+
+	@Override public void run() {
+		// until the following version, a user may still use the diff command to print test report differences.
+		// remove in next version
+		if ( goldenMasterPath.length == 1 ) {
+			logger.warn(
+					"The diff command for viewing a test report is deprecated and will be removed in the following version."
+							+ " Please use the new command 'show' to display a test report." );
+			final String[] args = { goldenMasterPath[0].toString() };
+			final Show c = new Show();
+			new CommandLine( c ).parseArgs( args );
+			c.run();
 			return;
 		}
+
+		final Path expectedSutStatePath = goldenMasterPath[0];
+		final Path actualSutStatePath = goldenMasterPath[1];
+
 		try {
-			final List<String> invalidFilters = FilterUtil.getInvalidFilters( exclude );
-			if ( !invalidFilters.isEmpty() ) {
-				FilterUtil.logWarningForInvalidFilters( invalidFilters );
-			} else {
-				printDiff();
+			if ( !hasValidExcludeOption() ) {
+				return;
 			}
+			final ActionReplayResult actionReplayResult =
+					createActionReplayResultFrom( expectedSutStatePath, actualSutStatePath );
+			final ActionReplayResult filteredResult = createFilteredActionReplayResult( actionReplayResult );
+			printGoldenMasterDifferences( filteredResult );
+			persistIfOutputOptionPresent( actionReplayResult );
 		} catch ( final Exception e ) {
 			ErrorHandler.handle( e );
 		}
 	}
 
-	private void printDiff() throws TestReportFormatException, IOException {
-		final TestReport report = TestReportUtil.load( testReport );
+	private boolean hasValidExcludeOption() {
```

Comment:
This seems copy paste from the other classes. Should we add a utility method within `FilterUtil`?

Reply:
Yes, it is from Show.java, I could extract it into FilterUtil and refactor Show.java and Diff.java accordingly.